src/Microsoft.OpenApi.Hidi/OpenApiService.cs

```diff
@@ -47,36 +49,24 @@ string filterbycollection
             {
                 if (string.IsNullOrEmpty(openapi) && string.IsNullOrEmpty(csdl))
                 {
-                    throw new ArgumentNullException("Please input a file path");
+                    throw new ArgumentException("Please input a file path");
                 }
-            }
-            catch (ArgumentNullException ex)
-            {
-                logger.LogError(ex.Message);
-                return;
-            }
-            try
-            {
                 if(output == null)
                 {
-                    throw new ArgumentException(nameof(output));
+                    throw new ArgumentNullException(nameof(output));
                 }
-            }
-            catch (ArgumentException ex)
-            {
-                logger.LogError(ex.Message);
-                return;
-            }
-            try
-            {
                 if (output.Exists)
                 {
-                    throw new IOException("The file you're writing to already exists. Please input a new file path.");
+                    throw new IOException($"The file {output} already exists. Please input a new file path.");
                 }
             }
-            catch (IOException ex)
+            catch (Exception ex)
             {
-                logger.LogError(ex.Message);
+#if DEBUG  
+                logger.LogCritical(ex, ex.Message);
+#else
+                logger.LogCritical(ex.Message);
+#endif
                 return;
```

Comment:
```suggestion
                return 1;
```
And update the method prototype to Task<int> so the exit code can be captured by system.commandline

Reply:
Do you mean `Task<int>`?

Reply:
yep